### PR TITLE
Add basic search based on a gazetteer to ANET

### PR DESCRIPTION
**Pull request by VassilIordanov**
This functionality allows users to search for geo names on a map, based on a geoSearcher service (ESRI, or others)

implements nci-agency/anet#999

### CODE REVIEW

**Comment by VassilIordanov** [comment]
will further document the `install.md` and `anet.yml.production.template` once I merge it in 2.1, to simplify the merging changes

File: anet.yml [hunk 1]
```diff
@@ -320,6 +320,9 @@ dictionary:
         zoomLevel: 10
       leafletOptions:
         attributionControl: false
+    geoSearcher:
+      provider: ESRI
+      url: "geocode.arcgis.com/arcgis/rest/services/World/GeocodeServer/find?"
```
Comment (gjvoosten) [inline anet.yml]: Should the question mark be part of the URL?

File: client/src/components/Leaflet.js [hunk 2]
```diff
@@ -21,6 +23,25 @@ const css = {
...
 
+class CustomURLESRIProvider extends EsriProvider {
+	endpoint({ query, protocol } = {}) {
+	  const { params } = this.options;
+	  const paramString = this.getParamString({
+		...params,
+		f: 'json',
+		text: query,
+	  })
+	  return `${protocol}//${Settings.imagery.geoSearcher.url}${paramString}`;
```
[the remark this answers is not in the document]
Comment (gjvoosten) [reply client/src/components/Leaflet.js]: Should the question mark be here instead (between url and paramString)?

File: client/src/components/Leaflet.js [hunk 3]
```diff
@@ -21,6 +23,25 @@ const css = {
...
+	}
+  }
+
+const geoSearcherProviders = {
+	ESRI : () => { return new CustomURLESRIProvider()},
+	OSM :  () => { return new OpenStreetMapProvider()}
+}
+
+const searchProvider = Settings.imagery.geoSearcher && geoSearcherProviders[Settings.imagery.geoSearcher.provider]()
```
Comment (gjvoosten) [inline client/src/components/Leaflet.js]: The lines above can be simplified to:
```
const geoSearcherProviders = {
	ESRI: CustomURLESRIProvider,
	OSM: OpenStreetMapProvider,
}

const searchProvider = Settings.imagery.geoSearcher && new geoSearcherProviders[Settings.imagery.geoSearcher.provider]()
```

File: client/src/components/Leaflet.js [hunk 4]
```diff
@@ -60,6 +81,10 @@ class BaseLeaflet extends Component {
 										 Settings.imagery.mapOptions.crs && { crs: CRS[Settings.imagery.mapOptions.crs] })
 		const map = new Map(this.mapId, mapOptions).setView( Settings.imagery.mapOptions.homeView.location,
 															 Settings.imagery.mapOptions.homeView.zoomLevel)
+		if (searchProvider) {
+		  new GeoSearchControl({ provider: searchProvider }).addTo(map)
```
Comment (gjvoosten) [inline client/src/components/Leaflet.js]: Tab-indent.

File: client/src/components/Leaflet.js [hunk 5]
```diff
@@ -21,6 +23,25 @@ const css = {
 	zIndex: 1,
 }
 
+class CustomURLESRIProvider extends EsriProvider {
```
Comment (gjvoosten) [inline client/src/components/Leaflet.js]: Change class to `CustomUrlEsriProvider` for readability.

File: client/src/components/Leaflet.js (hunk 3 above)
Comment (VassilIordanov) [reply client/src/components/Leaflet.js]: I would prefer to keep the factory as different parameters can be passed to the constructor. For instance, I should have taken the time and made the URL for ESRI to be one such parameter

**Pull request merged**